### PR TITLE
Skilling Calculators - Obsidian Redesign

### PR DESCRIPTION
**Pull request by psikoi**
Redesigned the skilling calculators, everything important should be listed on the commit message.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java [hunk 1]
```diff
@@ -58,7 +58,10 @@
 	public MaterialTabGroup(JPanel display)
 	{
 		this.display = display;
-		this.display.setLayout(new BorderLayout());
+		if (display != null)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java]: I would instead make second constructor

Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java]: Oh, good catch.

**Comment by psikoi** [comment]
- Added second constructor to the material tab group
- Refactored the IconTextField to include the FlatTextField

@deathbeam we good?

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java [hunk 2]
```diff
@@ -223,15 +248,15 @@ private void calculate()
...
+			slot.setText("Lvl. " + slot.action.level + " (" + formatXPActionString(xp, actionCount));
 			slot.setAvailable(currentLevel >= slot.action.level);
 			slot.value = xp;
 		}
 	}
 
 	private String formatXPActionString(double xp, int actionCount)
 	{
-		return XP_FORMAT.format(xp) + "xp - " + NumberFormat.getIntegerInstance().format(actionCount) + (actionCount > 1 ? " actions" : " action");
+		return XP_FORMAT.format(xp) + "exp) - " + NumberFormat.getIntegerInstance().format(actionCount) + (actionCount > 1 ? " actions" : " action");
```
Comment (sethtroll) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: remove the extra parenthesis

Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: Not extra. The string that comes before it has (

File: runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java [hunk 3]
```diff
@@ -67,35 +59,44 @@ public IconTextField()
...
+		textField = new FlatTextField();
+		textField.setBorder(null);
 
-		add(iconWrapperLabel, BorderLayout.WEST);
-		add(textField, BorderLayout.CENTER);
+		JTextField innerTxt = textField.getTextField();
+		innerTxt.removeMouseListener(innerTxt.getMouseListeners()[innerTxt.getMouseListeners().length - 1]);
 
-		textField.addMouseListener(new MouseAdapter()
+		MouseListener hoverEffect = (new MouseAdapter()
```
[the remark this answers is not in the document]
Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java]: Copy paste error. Damn

**Comment by psikoi** [comment]
Fixed two typos on the code:
- Accidental parenthesis on the mouse listener
- Accidental parenthesis (again) on the skill calc combined actions panel

**Pull request merged**